File: incomplete_fizz_buzz/ruby/lib/fizz_buzz.rb
module FizzBuzz
  module_function

  def fizz_buzz(number, printer) 
       printer.execute(check(number))
  end

  def fizz_buzz_history(number, printer)
        printer.execute("#{number}, #{check(number)}")
  end

  def write(histories)
    file = File.new('data.txt', 'w')
    for i in 0 .. (histories.size - 1) do
      history = histories[i]
        file.puts("#{history}, #{check(history)}")
    end
    file.close
  end
end

  def check(value)
    if value == 0 
        responce = value.to_s
    elsif value % 15 == 0
        responce = 'FizzBuzz'
    elsif value % 3 == 0
        responce = 'Fizz'
    elsif value % 5 == 0
        responce = 'Buzz'
    else 
        responce = value.to_s
    end
  end

File: incomplete_fizz_buzz/java/src/FizzBuzz.java
public class FizzBuzz {
  public static void fizzBuzz(int number, Printable printer) {
    printer.execute(fizzBuzz(number));
  }

  public static String fizzBuzz(int number) {
    if(number == 0) {
      return String.valueOf(number);
    } else if(number % 15 == 0) {
      return "FizzBuzz";
    } else if(number % 3 == 0) {
      return "Fizz";
    } else if(number % 5 == 0) {
      return "Buzz";
    } else {
      return String.valueOf(number);
    }
  }
}
